# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

if pkg-config --exists python2-build; then
  REQUIRES="$REQUIRES python2-build"
elif pkg-config --exists python3-build; then
  REQUIRES="$REQUIRES python3-build"
else
  echo Python 2 or Python 3 packages required; exit 1
fi

# configure (ac-2.69; am-1.13; lt-2.4.2)
libtoolize -c -f; aclocal; autoheader; autoconf; automake -a -c
./configure --disable-static --prefix=

# build
make

# install
install -d $DESTDIR/include/libevdev-1.0/libevdev
install -m 644 libevdev/libevdev.h $DESTDIR/include/libevdev-1.0/libevdev
install -m 644 libevdev/libevdev-uinput.h $DESTDIR/include/libevdev-1.0/libevdev
install -d $DESTDIR/lib
install libevdev/.libs/libevdev.so.1.* $DESTDIR/lib/libevdev.so.1
ln -sf libevdev.so.1 $DESTDIR/lib/libevdev.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 libevdev.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/libevdev-build.pc << EOF
Name: libevdev
Version: 0.5
Description: wrapper library for event device interface
Requires: $REQUIRES

devel=\\
/include/libevdev-1.0/libevdev/libevdev.h \\
/include/libevdev-1.0/libevdev/libevdev-uinput.h \\
/lib/libevdev.so \\
/lib/pkgconfig/libevdev.pc

exec=\\
/lib/libevdev.so.1
EOF

exit

# patch
--- libevdev-0.5.orig/configure.ac
+++ libevdev-0.5/configure.ac
@@ -32,6 +32,7 @@
 AC_USE_SYSTEM_EXTENSIONS
 
 AM_INIT_AUTOMAKE([1.11 foreign no-dist-gzip dist-xz])
+AM_MAINTAINER_MODE
 
 # Before making a release, the LIBEVDEV_LT_VERSION string should be
 # modified.

# source
https://freedesktop.org/software/libevdev/libevdev-0.5.tar.xz
